server/sonar-server/src/test/java/org/sonar/server/updatecenter/ws/UploadActionTest.java

```diff
@@ -53,79 +55,82 @@
   public UserSessionRule userSession = UserSessionRule.standalone();
 
   DefaultServerFileSystem fileSystem = mock(DefaultServerFileSystem.class);
-  File downloadedPluginDir;
-  File plugin = new File(Resources.getResource(getClass(), "UploadActionTest/plugin.jar").getFile());
+  File pluginDirectory;
+
+  File plugin = new File(getClass().getResource("UploadActionTest/plugin.jar").getFile());
+  WsActionTester underTest;
 
   @Before
   public void setUp() throws Exception {
-    downloadedPluginDir = folder.newFolder();
-    when(fileSystem.getDownloadedPluginsDir()).thenReturn(downloadedPluginDir);
+    pluginDirectory = folder.newFolder();
+    when(fileSystem.getDownloadedPluginsDir()).thenReturn(pluginDirectory);
+    underTest = new WsActionTester(new UploadAction(userSession, fileSystem));
   }
 
   @Test
   public void upload_plugin() throws Exception {
```

Comment:
maybe you can test that the input stream is closed after the WS call? it's the typical source of error that are hard to track